packages/roosterjs-content-model/lib/publicApi/utils/selectWord.ts

```diff
@@ -0,0 +1,103 @@
+import { ContentModelBlockGroup } from '../../publicTypes/group/ContentModelBlockGroup';
+import { ContentModelSegment } from '../../publicTypes/segment/ContentModelSegment';
+import { ContentModelText } from '../../publicTypes/segment/ContentModelText';
+import { createText } from '../../modelApi/creators/createText';
+
+export function selectWord(
+    path: ContentModelBlockGroup[],
+    marker: ContentModelSegment
+): ContentModelSegment[] {
+    const parent = path[0];
+    const block = parent.blocks[0];
+
+    if (block.blockType == 'Paragraph') {
+        const segments: ContentModelSegment[] = [];
+        let markerSelectionIndex = block.segments.indexOf(marker);
+        for (let i = markerSelectionIndex - 1; i >= 0; i--) {
+            const currentSegment = block.segments[i];
+            if (currentSegment.segmentType == 'Text') {
+                const found = findDelimiter(currentSegment, false);
+                if (found != null) {
+                    if (found == currentSegment.text.length) {
+                        break;
+                    }
```

Comment:
Make it return -1 as "not found", so only need to check once here

Reply:
We still need to check twice. The first check is to see if a delimiter was found and the second to dismiss it if the delimiter was found at the very end of the string.

Reply:
What is the difference between found in middle and found at the end? My understanding is, as long as found, we just need to split after it, no matter where it is.

Reply:
Here we are searching right to left, so if the first thing you find is the delimiter character, it means that segment does not need to be split because it already is.

So for example: "this is a short sentence". If you put the selection marker right before the s in short, the segment is already split. The 3 segments are: "this is a ", SelectionMarker,"short sentence". We only need to split the right one so we can stop checking the left.

Reply:
I see. That makes sense to me.

Although without the second check, I believe it can still work even we have an extra split. (It is ok to have an empty text node in content model because later it will be removed when normalize), but I think it is ok to have this optimization here as you suggested.